packages/block-library/src/latest-posts/index.php

```diff
@@ -173,3 +173,34 @@ function register_block_core_latest_posts() {
 	);
 }
 add_action( 'init', 'register_block_core_latest_posts' );
+
+/**
+ * Handles outdated versions of the `core/latest-posts` block by converting
+ * attribute `categories` from a numeric string to an array with key `id`.
+ *
+ * This is done to accommodate the changes introduced in #20781 that sought to
+ * add support for multiple categories to the block. However, given that this
+ * block is dynamic, the usual provisions for block migration are insufficient,
+ * as they only act when a block is loaded in the editor.
+ *
+ * TODO: Remove when and if the bottom client-side deprecation for this block
+ * is removed.
+ *
+ * @param array $block A single parsed block object.
+ *
+ * @return array The migrated block object.
+ */
+function block_core_latest_posts_migrate_categories( $block ) {
+	if (
+		'core/latest-posts' === $block['blockName'] &&
+		isset( $block['attrs']['categories'] ) &&
```

Comment:
I believe it may not be safe to assume `$block['attrs']` is set, given that in the implementation of `render_block`, its presence is checked _after_ the `render_block_data` filter we're hooking here.

https://github.com/WordPress/wordpress-develop/blob/90b0f49/src/wp-includes/blocks.php#L481-L483

I'm not actually sure when (or if) this can occur, since as best I can tell from the implementation of the block parsing, it always attempts to assign a value for a block:

https://github.com/WordPress/wordpress-develop/blob/90b0f49/src/wp-includes/class-wp-block-parser.php#L440-L446

A simple alternative to consider is `empty`, which is unique in that it will happily accept any input (e.g. `empty( $block['this']['definitely']['doesnt']['exist'] )`). Assuming we're only wanting to deal with a non-empty-string value of `categories`, this could be:

```suggestion
		! empty( $block['attrs']['categories'] ) &&
```